### ISSUE TITLE
`Logger.forEnclosingClass()` needed for using in static Java methods

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/logging#33]
Having `WithLogging` interface is not enough even for our needs. We have Java code, fortunately, not much of it, which is static utilities. There we cannot do `WithLogging` because there are not instances of such a class in a static context.

Now we have to do something like this:

```java
    private static final Logger<?> logger = LoggingFactory.getLogger(getKotlinClass(Linker.class));
```

It would be good to have something similarly to what Flogger offers for such cases:

```java
    private static final Logger<?> logger = Logger.forEnclosingClass();
```
I'm not sure that it would be exactly as above in our multiplatform arrangement. But the following code will also looks not so bad to me:
```java
    private static final Logger<?> logger = LoggingFactory.forEnclosingClass();
```

**Comment by alexander-yevsyukov** [comment]
@yevhenii-nadtochii, after spending some time migrating `base`, I think we need this convenience for moving on faster.

**Comment by alexander-yevsyukov** [comment]
@armiol, FYI in the context of [this PR in `base`](https://github.com/SpineEventEngine/base/pull/802).

**Comment by yevhenii-nadtochii** [comment]
Addressed in #36.
